calculator/main.py

```diff
@@ -58,18 +89,39 @@ def get_history(calc):
         ie:
         ('2016-05-20 12:00:00', 'add', (1, 2), 3),
     """
-    pass
+    return calc['history']
 
 
 def reset_history(calc):
     """
     Resets the calculator history back to an empty list.
     """
-    pass
+    calc['history'] = []
+    return calc
 
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    #'perform_operation(calc, operation, params)'
+    if calc['history'] != []:
+        operation = calc['history'][-1][1]
+        params = calc['history'][-1][2]
+        operation = calc['operations'][operation]
```

Comment:
`perform_operations()` takes the key for the operation it is running, not the value for that key, so this line should be left out.